cmd/main.go

```diff
@@ -2,43 +2,56 @@ package main
 
 import (
 	"encoding/json"
+	"flag"
 	"fmt"
 	"os"
-	"strings"
 	"time"
 
 	"github.com/duneanalytics/duneapi-client-go/config"
 	"github.com/duneanalytics/duneapi-client-go/dune"
 )
 
 func main() {
-	cfg, err := config.ParseConfig()
-	if err != nil {
-		if !strings.HasPrefix(err.Error(), "Usage") {
-			os.Exit(1)
-		}
-		return
-	}
-	if (cfg.ExecutionID == "" && cfg.QueryID == 0) || (cfg.ExecutionID != "" && cfg.QueryID != 0) {
+	queryID := flag.Int("q", 0, "The ID of the query to execute. Conflicts with -e")
+	queryParametersStr := flag.String("p", "{}", "Parameters to pass to the query in JSON format")
+	executionID := flag.String("e", "", "ID of an existing execution to check status. Conflicts with -q")
+	maxRetries := flag.Int("max-retries", 5, "Max number of get errors tolerated before giving up")
+	pollInterval := flag.Duration("poll-interval", 5*time.Second, "Interval in seconds for polling for results")
+
+	flag.Parse()
+
+	if (*executionID == "") == (*queryID == 0) {
```

Comment:
nit: clever but makes it difficult to read for the reader (or at least my small brain)

Reply:
:smile: I previously had this as `(cfg.ExecutionID == "" && cfg.QueryID == 0) || (cfg.ExecutionID != "" && cfg.QueryID != 0)`, and it annoyed me that it's so long. But I guess you're right, readability over simplicity.